From 134f44d6debd4ccf5510488f78bfc0f7fe850eff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Mon, 26 Aug 2019 08:42:47 +0200
Subject: [PATCH] Test a regression in a lookbehind after a condition
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The regression is caused with this upstream commit:

commit d70442f2c962ca816b8d03504a07e618c271a9fe
Author: ph10 <ph10@6239d852-aaf2-0410-a92c-79f79f948069>
Date:   Tue Jul 16 15:06:21 2019 +0000

    Fix lookbehind within lookahead within lookbehind misbehaviour bug.

    git-svn-id: svn://vcs.exim.org/pcre2/code/trunk@1133 6239d852-aaf2-0410-a92c-79f79f948069

https://bugs.exim.org/show_bug.cgi?id=2433
https://bugzilla.redhat.com/show_bug.cgi?id=1743863
Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 testdata/testinput2  | 3 +++
 testdata/testoutput2 | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/testdata/testinput2 b/testdata/testinput2
index 9ac8646..2dc99da 100644
--- a/testdata/testinput2
+++ b/testdata/testinput2
@@ -5605,4 +5605,7 @@ a)"xI
 /\A\s*((?:[^`]{28500}){4}|a)/I
     a
 
+/^(?<A>a)(?(<A>)b)((?<=b).*)$/
+abc
+
 # End of testinput2
diff --git a/testdata/testoutput2 b/testdata/testoutput2
index bcc1aae..3875e1e 100644
--- a/testdata/testoutput2
+++ b/testdata/testoutput2
@@ -16975,6 +16975,12 @@ Subject length lower bound = 1
  0: a
  1: a
 
+/^(?<A>a)(?(<A>)b)((?<=b).*)$/
+abc
+ 0: abc
+ 1: a
+ 2: c
+
 # End of testinput2
 Error -70: PCRE2_ERROR_BADDATA (unknown error number)
 Error -62: bad serialized data
-- 
2.21.0

